crates/subspace-core-primitives/src/objects.rs

```diff
@@ -129,50 +129,55 @@ pub struct PieceObjectMapping {
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Ord, PartialOrd, Hash, Encode, Decode, TypeInfo)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
-pub enum GlobalObject {
-    /// V0 of object mapping data structure
-    #[codec(index = 0)]
-    #[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
-    V0 {
-        /// Piece index where object is contained (at least its beginning, might not fit fully).
-        /// The index and offset must be first, so that the Ord derives are valid.
-        piece_index: PieceIndex,
-        /// Offset of the object in that piece
-        offset: u32,
-        /// Object hash
-        hash: Blake3Hash,
-    },
+pub struct GlobalObject {
+    /// Object hash.
+    /// We order by hash, so object hash lookups can be performed efficiently.
+    hash: Blake3Hash,
+    /// Piece index where object is contained (at least its beginning, might not fit fully)
+    piece_index: PieceIndex,
+    /// Offset of the object in that piece
+    offset: u32,
 }
 
 impl GlobalObject {
+    /// Returns a newly created GlobalObject from its fields.
+    pub fn new(hash: Blake3Hash, piece_index: PieceIndex, offset: u32) -> Self {
+        Self {
+            hash,
+            piece_index,
+            offset,
+        }
+    }
+
+    /// Object hash
+    pub fn hash(&self) -> Blake3Hash {
+        self.hash
+    }
+
     /// Piece index where object is contained (at least its beginning, might not fit fully)
     pub fn piece_index(&self) -> PieceIndex {
-        match self {
-            Self::V0 { piece_index, .. } => *piece_index,
-        }
+        self.piece_index
     }
 
     /// Offset of the object
     pub fn offset(&self) -> u32 {
-        match self {
-            Self::V0 { offset, .. } => *offset,
-        }
-    }
-
-    /// Object hash
-    pub fn hash(&self) -> Blake3Hash {
-        match self {
-            Self::V0 { hash, .. } => *hash,
-        }
+        self.offset
     }
 }
 
 /// Mapping of objects stored in the history of the blockchain
-#[derive(Debug, Default, Clone, PartialEq, Eq, Ord, PartialOrd, Hash, Encode, Decode, TypeInfo)]
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Hash, Encode, Decode, TypeInfo)]
```

Comment:
Why was `Default` removed here? Seems useful to be able to create an empty one in tests and such.

Reply:
```rust
impl Default for GlobalObjectMapping {
    fn default() -> Self {
        Self::V0 {
            objects: Vec::new()
        }
    }
}
```